notebooks/04-Conditionals.ipynb

```diff
@@ -0,0 +1,1275 @@
+   "metadata": {
+    "slideshow": {
+     "slide_type": "fragment"
+    }
+   },
+   "source": [
+    "* Benefits include:\n",
+    "  * Error checking\n",
+    "  * Logic checking\n",
+    "  * Automation"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "slideshow": {
+     "slide_type": "slide"
+    }
+   },
+   "source": [
+    "## Program Control"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "slideshow": {
+     "slide_type": "slide"
+    }
+   },
+   "source": [
+    "### `if` Statements"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "slideshow": {
+     "slide_type": "fragment"
+    }
+   },
+   "source": [
+    "* `if` statement syntax in Python isn't quite as easy as our examples, but it's pretty close..."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "slideshow": {
+     "slide_type": "fragment"
+    }
+   },
+   "source": [
+    "* `if` statements begin with the `if` keyword and are followed by a logical test and a colon"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "slideshow": {
+     "slide_type": "-"
+    }
+   },
+   "source": [
+    "   ```python\n",
+    "   if tired == 'Yes':\n",
+    "   ```"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "slideshow": {
+     "slide_type": "slide"
+    }
+   },
+   "source": [
+    "* The then statement directly follows the `if` statement as an indented block"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "slideshow": {
+     "slide_type": "-"
+    }
+   },
+   "source": [
+    "   ```python\n",
+    "   if tired == 'Yes':\n",
+    "       go_to_bed()\n",
+    "   ```"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "slideshow": {
+     "slide_type": "fragment"
+    }
+   },
+   "source": [
+    "* Note that the actual word **then** is never used - it's implied by the indent\n",
+    "* Interpret as:\n",
+    "\n",
+    "   ```python\n",
+    "   if condition_holds:\n",
+    "       execute_code()\n",
+    "   ```"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "slideshow": {
+     "slide_type": "slide"
+    }
+   },
+   "source": [
+    "* The implied then-block can have as many lines as necessary"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "slideshow": {
+     "slide_type": "-"
+    }
+   },
+   "source": [
+    "   ``` python\n",
+    "   if tired == 'Yes':\n",
+    "       brush_teeth()\n",
+    "       turn_off_lights()\n",
+    "       go_to_bed()\n",
+    "   ```"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "slideshow": {
+     "slide_type": "fragment"
+    }
+   },
+   "source": [
+    "* All of the indented code will execute only if `tired == 'Yes'` is `True`"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "slideshow": {
+     "slide_type": "slide"
+    }
+   },
+   "source": [
+    "### Non psuedo code"
```

Comment:
Go for the double hyphen!

Reply:
It's always been a dream of mine

Reply:
Honestly I'm torn on which side should be single and which double

Reply:
Somehow "psuedo" is misspelled and I never saw it until I opened the notebook json in Vim.